### ISSUE TITLE
Add Documentation

### ISSUE THREAD

**Issue by MarkKoz** [tophattwaffle/BotHATTwaffle#9]
This issue tracks the goal of documenting code to improve maintainability. 

* [XML Documentation Comments](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/xml-documentation-comments) should be used for objects and members. 
    * The [`summary`](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/summary) should be used to clarify _functionality_, not implementation details or miscellaneous information, which should be explored in [`remarks`](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/remarks) when needed.
    * Include a short, one-line summary as the first line of `summary`. When applicable, include a more detailed summary as a [`para`](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/para) within the `summary`, using multiple paragraphs as needed.
    * Use [`cref`](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/cref-attribute), [`paramref`](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/paramref), and [`typeparamref`](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/typeparamref), respectively, when referencing code in documentation comments.
    * Properties and fields should generally be self-evident due to their names, but documentation comments for them are also welcome.
    * Returning a `Task` without a value should be documented as
        > No object or value is returned by this method when it completes.
* Clarify the purpose of any local fields with discretion.
* Comment statements with discretion and the goal of minimising the need to reference other documentation (external or internal) or analyse code at another location. Self-evident statements _do not_ need to be commented as the comments would be redundant.
* Documenting commands using the `Summary` and `Remarks` attributes should generally be sufficient; revise/format the attributes with discretion.

Progress:
- Commands
    - [x] `HelpModule`
    - [x] `InformatonModule`
    - [ ] `ModerationModule`
    - [ ] `PlaytestingModule`
    - [x] `ShadersModule`
    - [x] `ToolsModule`
    - [x] `UtilitiesModule`
- [x] Extensions
- Models
    - [x] `UserData`
- Services
    - Download
        - [ ] `Downloader`
        - [ ] `DownloadService`
        - [ ] `FtpDownloader`
        - [ ] `SftpDownloader`
    - Playtesting
        - [x] `EventCalendarService`
        - [ ] `PlaytestingService`
    - [ ] `DataService`
    - [ ] `MessageListener`
    - [x] `MuteService`
    - [x] `TimerService`
- [ ] `ConsoleCopy`
- [x] `Program`

**Comment by misyltoad** [comment]
If you make your code self-descriptive and self-documenting you don't need to add summaries really, it's not a library. Scrolling past big green blocks gets *really* irritating after a while.

**Comment by MarkKoz** [comment]
I agree with it being unnecessary in this context, especially given how little attention this project will ever get.

From experience, I hate working with poorly documented code. Even if the code is self-descriptive and self-documenting, it's never perfect. Regardless, explicit documentation adds a layer of security. I do think it does more good than harm. In most cases, documentation is quicker and more effective at describing code than the code itself. Exceptions to this may be very short or basic sections of code, but I document them anyway for consistency's sake.

One gripe I have is that documentation has to be revised when changes are made, but there's no problem with postponing that for some other future commit. Another is that C# unfortunately uses XML, which is definitely one of the more verbose and ugly syntaxes that I've seen for documentation. Requiring documentation may also deter others from contributing. However, we (well, at least I) don't plan on being strict on pull requests in regards to style or documentation compliance due to the very low volume. As for scrolling past big green blocks of text, I feel you, but there are ways of remedying that.
